### ISSUE TITLE
Allow user to select separator and delimiter convention for EUR currency

### ISSUE THREAD

**Issue by zachgoll** [maybe-finance/maybe#1154]
### Discussed in https://github.com/maybe-finance/maybe/discussions/1093

<div type='discussions-op-text'>

<sup>Originally posted by **825i** August 15, 2024</sup>
### Describe the feature

In most areas of the world numbers are written like so:

2,312.25

Not:

2.312,25

That is, comma first, period second.

There should be at the very least an option to reverse the placement.

Further reasoning for this is that my debts to me looks like I only owe €25.32 cents. Not €25,325...

![image](https://github.com/user-attachments/assets/5de0d162-9acf-449b-baf9-343c9b6f14cb)

### Why is this feature important?

There is the possibility of misunderstanding balances.

### Additional context, screenshots, and relevant links

Comma and period should be replaced eg. in the case of all these amounts.

![Screenshot from 2024-08-15 10-02-27](https://github.com/user-attachments/assets/822dc8d4-0647-466c-a110-7e1d824f2076)</div>

**Comment by tonyvince** [comment]
> Which countries predominantly view EUR with, as the thousands delimiter? (i.e. €25,000.50)
Ireland, Cyprus, and Malta use `,` as a thousand separator and `.` as a decimal separator like the rest of the (sane) world. 

> From what I've seen, the predominant way is €25.000,50?
Yes.

**Comment by gariasf** [comment]
> Which countries predominantly view EUR with, as the thousands delimiter? (i.e. €25,000.50)

In Spain, local banks and other financial platforms generally use `,` as a decimal separator and `.` as the thousand separator. With that said, plenty of other foreign platforms still use the other approach. 

What matters is the precision of the decimal amount. It is rarely seen with less than three digits, perhaps except for crypto, and I would say that's where the confusion comes from. Normally, we would see a thousand separator with 4 digits, three of them after the separator, whichever it is. Hence, the confusion when seeing three decimal numbers on the decimal part. 
I'd say that if we can only display the decimals rounded to two digits of precision everywhere sensible, this topic would become a non issue. Otherwise, if it were necessary, a legend, or format indication should be shown. 

My take: Trying to account for this issue should not be relevant before the stable release. Moreover, it might also add unnecessary complexity and friction for other features. I'd leave it for polishing at a later stage.

**Comment by zachgoll** [comment]
@gariasf thanks for the detail!  So we do have a currencies file that gives us a default precision:

https://github.com/maybe-finance/maybe/blob/730e58d763b223cc5c21a9ebb464274be3c819a3/config/currencies.yml#L14

This takes care of formatting the correct number of decimals.

That said, I agree that this is "polish" that will likely be deferred until after we get the hosted app up and running (although open to contributions!)

**Comment by lasconic** [comment]
In France, it's `25 000,50 €`. I checked 3 different banking apps.

**Comment by vanhalt** [comment]
Just to confirm, based on what I read, adding a dependency like [RubyMoney](https://github.com/RubyMoney) is not an option, right?

**Comment by zachgoll** [comment]
@vanhalt correct, our current `Money` implementation is somewhat similar to RubyMoney, but has some additional customizations and most importantly, has an API that deals in _whole monetary values_ rather than cents.  RubyMoney and the Rails integration for it have an API that is highly coupled to the idea of storing "cents" in the database which was a bit confusing in our implementation given we're storing whole numbers in the DB.  In writing our own, we were optimizing for dependency reduction + API simplicity.

Furthermore, RubyMoney deals with all formatting _internally_, while our `Money` class delegates the formatting to the built-in Rails helper, `number_to_currency`

To fix this, we will likely need a "Ruleset" (similar to what RubyMoney has) that knows how to generate these options based on the user's provided locale (which will be possible to set in their preferences):

https://github.com/maybe-finance/maybe/blob/edbf4eb3d6e7a4ca896fe536c957c2311ca7d51b/lib/money.rb#L100-L108
